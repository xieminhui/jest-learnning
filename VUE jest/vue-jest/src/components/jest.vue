<template>
   <div>
      <MessageList :messages="messages"></MessageList>
   </div>
</template>

<script>
import MessageList from './MessageList'

export default {
  name: 'jest',
  data () {
    return {
      messages: ['Hey John', 'Howdy Paco']
    }
  },
  components: {
    MessageList
  }
}
</script>
