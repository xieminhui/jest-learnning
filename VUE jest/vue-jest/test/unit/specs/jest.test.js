import Vue from 'vue'
import jest from '../../../src/components/jest.vue'

describe('jest.test.js', () => {
  let Cmp, vm
  beforeEach(() => {
    Cmp = Vue.extend(jest)
    vm = new Cmp({
      data: {
        messages: ['cat']
      }
    }).$mount()
  })

  it('equals messages to ["cat"]', () => {
    expect(vm.messages).toEqual(['cat'])
  })

  it('has the expected html structure', () => {
    expect(vm.$el).toMatchSnapshot();
  })
})
