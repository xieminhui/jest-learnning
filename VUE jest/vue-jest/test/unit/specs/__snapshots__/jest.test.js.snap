// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest.test.js has the expected html structure 1`] = `
<div>
  <ul>
    <li>
      
      cat
   
    </li>
  </ul>
</div>
`;
